stackslib/src/chainstate/stacks/boot/pox_4_tests.rs

```diff
@@ -5209,25 +5266,39 @@ fn stack_stx_signer_key() {
         .block_height_to_reward_cycle(block_height)
         .unwrap();
     let reward_cycle_ht = burnchain.reward_cycle_to_block_height(next_reward_cycle);
-    let mut reward_set = get_reward_set_entries_at(&mut peer, &latest_block, reward_cycle_ht);
-    assert_eq!(reward_set.len(), 1);
-    let reward_entry = reward_set.pop().unwrap();
-    assert_eq!(
-        PoxAddress::try_from_pox_tuple(false, &pox_addr_val).unwrap(),
-        reward_entry.reward_address
-    );
+    let reward_set = get_reward_set_entries_at(&mut peer, &latest_block, reward_cycle_ht);
+    assert_eq!(reward_set.len(), {
```

Comment:
Why is there a discrepancy?  Can you document it?

Reply:
Yep I can add a comment - basically, in 3.0 there is a default signer (separate from the accounts used in the test) to ensure that Nakamoto can progress.